addons/mail/static/src/core/common/composer.js

```diff
@@ -803,13 +811,29 @@ export class Composer extends Component {
         }
     }
 
-    restoreContent() {
+    async restoreContent() {
         const composer = toRaw(this.props.composer);
         try {
             const config = JSON.parse(browser.localStorage.getItem(composer.localId));
             if (config.text) {
                 composer.emailAddSignature = config.emailAddSignature;
                 composer.text = config.text;
+                await this.suggestion.suggestionService.fetchPartners(
+                    config.mentionedPartners.filter(
+                        (m) => !this.store.Persona.get(m)
+                    ).map((m) => m.id),
+                    this.thread
+                );
+                for (const mentionedPartner of config.mentionedPartners) {
+                    this.props.composer.mentionedPartners.push(
+                        this.store.Persona.get(mentionedPartner)
+                    );
+                }
```

Comment:
This should also not just rely on `get`, but actually go through the _conditions_ of the suggestion service to determine if the partner is acceptable (`searchPartnerSuggestions` in particular, or a new function to have common logic, as sorting can be omitted here).